plenum/server/replica.py

```diff
@@ -1000,6 +1005,7 @@ def processPrepare(self, prepare: Prepare, sender: str) -> None:
                 self.stats.inc(TPCStat.PrepareRcvd)
                 self.logger.debug("{} processed incoming PREPARE {}".format(
                     self, (prepare.viewNo, prepare.ppSeqNo)))
+                self._setup_for_non_master(is_catchup=False)
```

Comment:
Why do we need to call it here? Will it be enough if we call it on the beginning of `processPrePrepare` only?
The main thing that this method does is re-calculation of last_ordered, which is needed to make suer that PrePrepare is in order.

Reply:
After catch up we can have last_ordered_3pc = 0. _setup_for_non_master there corresponds to logic of usage it in processPrePrepare: collect last_order after all message responce. But I'm not sure in it because it can be reason for more message requests that we need.